packages/grafana-ui/src/components/ToggleButtonGroup/ToggleButtonGroup.tsx

```diff
@@ -0,0 +1,186 @@
+import React, { PureComponent, useContext, FC } from 'react';
+import { Tooltip } from '../Tooltip/Tooltip';
+import { ThemeContext } from '../../themes/index';
+import { css, cx } from 'emotion';
+import { GrafanaTheme } from '../../types/theme';
+import tinycolor from 'tinycolor2';
+import { ToggleButtonGroupProps, ToggleButtonProps, ToggleButtonState } from './types';
+import { increaseContrast } from '../../utils/colors';
+import { ButtonSize } from '../Button/AbstractButton';
+
+const ToggleButtonGroup: FC<ToggleButtonGroupProps> = ({ label, children, propsPriority }) => {
+  const styles = getStyles(useContext(ThemeContext));
+  return (
+    <div>
+      <span className={styles.label}>{label}</span>
+      {React.Children.map(children, (child: any, index) => {
+        // Set correct style depending on first/middle/last
+        const style = index === 0 ? styles.first : index === children.length - 1 ? styles.last : styles.middle;
+
+        return React.cloneElement(child, {
+          className: style,
+          propsPriority,
```

Comment:
This seems suspicious... ;) Can you elaborate?

Reply:
It is used to set styles for the first and last child. Should probably set it with CSS `:first-child` and `:last-child` instead, but I wrote it before I had properly figured out emotion.

`propsPriority` is used to turn off the internal `selected` state in `ToggleButton`. This sets the prop for all buttons in the group.